src/test/MavenTestDevModeActions.ts

```diff
@@ -99,8 +99,10 @@ it('Run tests for sample maven project', async () => {
 it('start maven with options from liberty dashboard', async () => {      
     
   const reportPath = path.join(utils.getMvnProjectPath(),"target","site","failsafe-report.html");
-  const deleteReport = await utils.deleteReports(reportPath);
-  expect (deleteReport).to.be.true;
+  const alternateReportPath = path.join(utils.getMvnProjectPath(), "target", "reports", "failsafe.html"); // new path to scan for the reports 
+  let deleteReport = await utils.deleteReports(reportPath);
+  let deleteAlternateReport = await utils.deleteReports(alternateReportPath);
+  expect (deleteReport || deleteAlternateReport).to.be.true; // there should be a report available irrespective of the surefire versions we use , so either one needs to be true 
```

Comment:
I believe the intention behind deleting the reports here is to set up the test so that we can confirm that the report files are created later. That means that _both_ reports should be deleted or not exist at this point. Instead of an OR, this should be an AND.

Reply:
The reports can only be either in the directory 'reports' or in 'site' depending upon the version of the plugin we use , if  AND is used then it expects report in 'reports' and 'site' to be deleted  which will not happen as the report will only be in one location at this point. if its OR then it will be only expecting to delete the report that is available either from 'reports' or 'site' .

Reply:
OK, let's take a step back and look at the `deleteReports()` function and what it is doing:
https://github.com/JoshwinThomasIBM/liberty-tools-vscode/blob/issue340fix/src/test/utils/testUtils.ts#L66-L81

- The first thing it does is check to see if `reportPath` exists. If it does _not_ exist, the method returns `true`, since the report is not there. The report does not have to be deleted for this function to return `true`. It can just not exist.
- If `reportPath` does exist, the function attempts to delete the file at the location of `reportPath`. If deletion succeeds, it returns `true` and if deletion fails, it returns `false`.

So now going back to the new code added. Let's assume that the tests are using the latest LMP and that test reports are being generated to the "reports" dir. The value of `deleteReport` after calling `deleteReports()` and passing in the "site" dir path will be `true`, since the report does not exist at the "site" dir.  Now, for `deleteAlternateReport`, the code calls `deleteReports()` and passes the "reports" dir path. Let's assume there is a problem with the filesystem and deleting the report fails. That means `deleteReports()` would return `false`.

On the next line where the test runs `expect`, we would expect the test to fail. The test is invalid if we continue on and a test report already exists. However, using an OR statement will not fail the test on this line and the test will run in an invalid state. On the other hand, using an AND would catch this problem and fail the test on this line as expected.

Essentially, because the `deleteReports()` method returns `true` if the report is deleted or did not exist in the first place, we should be using an AND operator. We want both reports to not exist when going into these tests.

Reply:
Understood , the changes are made as suggested